main.go

```diff
@@ -557,6 +559,14 @@ func (c *controller) sync(ctx context.Context) {
 		c.reconcileErrors.WithLabelValues(save).Inc()
 		level.Error(c.logger).Log("msg", "failed to save hashrings")
 	}
+
+	// If enabled, annotate pods with config hash on change. This should
+	// update the configmap inside the pod instantaneously as well, as
+	// opposed to having to wait kubelet sync period + cache (see
+	// https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#mounted-configmaps-are-updated-automatically)
+	if c.options.annotatePodsOnChange {
```

Comment:
what happens if the c.saveHashring func above errors? should we avoid annotating in that case?

Reply:
Yes, that's a good point, I'll make it conditional on no `err`.